lib/Sabre/Album/AlbumsHome.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Robin Appelman <robin@icewind.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Photos\Sabre\Album;
+
+use OCA\Photos\Album\AlbumMapper;
+use OCA\Photos\Album\AlbumWithFiles;
+use OCP\Files\Folder;
+use OCP\Files\IRootFolder;
+use OCP\IUser;
+use Sabre\DAV\Exception\Forbidden;
+use Sabre\DAV\Exception\NotFound;
+use Sabre\DAV\ICollection;
+
+class AlbumsHome implements ICollection {
+	private AlbumMapper $albumMapper;
+	private array $principalInfo;
+	private IUser $user;
+	private IRootFolder $rootFolder;
+	private Folder $userFolder;
+
+	public function __construct(
+		array $principalInfo,
+		AlbumMapper $albumMapper,
+		IUser $user,
+		IRootFolder $rootFolder
+	) {
+		$this->principalInfo = $principalInfo;
+		$this->albumMapper = $albumMapper;
+		$this->user = $user;
+		$this->rootFolder = $rootFolder;
+		$this->userFolder = $rootFolder->getUserFolder($user->getUID());
+	}
+
+	public function delete() {
+		throw new Forbidden();
+	}
+
+	public function getName(): string {
+		return 'albums';
+	}
+
+	public function setName($name) {
+		throw new Forbidden('Permission denied to rename this folder');
+	}
+
+	public function createFile($name, $data = null) {
+		throw new Forbidden('Not allowed to create files in this folder');
+	}
+
+	public function createDirectory($name) {
+		$uid = $this->user->getUID();
+		$this->albumMapper->create($uid, $name);
+	}
+
+	public function getChild($name) {
+		foreach ($this->getChildren() as $child) {
+			if ($child->getName() === $name) {
```

Comment:
I am right if I say that two albums will not be able to have the same name ? It makes sense in a DAV API, but for the user it might be confusing. Maybe we should distinguish between name and display name. But thinking about it I am not so sure.

@jancborchardt or @nimishavijay, should the user be able to have two albums with the same name ?

Reply:
I'd say no as it makes things more complicated (can of worms) for not much benefit